client/ayon_blender/api/plugin_load.py

```diff
@@ -0,0 +1,216 @@
+    # everything contained in the loaded collection.
+    context.view_layer.objects.active = loaded_objects[0]
+
+    from .plugin import create_blender_context  # todo: move import
+    operator_context = create_blender_context(
+        active=loaded_objects[0],
+        selected=loaded_objects
+    )
+
+    # https://blender.stackexchange.com/questions/289245/how-to-make-a-blender-library-override-in-python  # noqa
+    # https://docs.blender.org/api/current/bpy.types.ID.html#bpy.types.ID.override_hierarchy_create  # noqa
+    if bpy.app.version[0] >= 4:
+        with bpy.context.temp_override(**operator_context):
+            loaded_collection.override_hierarchy_create(
+                scene, context.view_layer, do_fully_editable=True
+            )
+    # Pre 4.0 method:
+    else:
+        pass
+
+    scene.collection.children.unlink(loaded_collection)
+
+    local_collection = get_local_collection(
+        overridden_collections,
+        loaded_collection,
+    )
+    return local_collection
+
+
+def get_local_collection(
+    overridden_collections: list[bpy.types.Collection],
+    loaded_collection: bpy.types.Collection,
+) -> bpy.types.Collection:
+    """Get the local (overridden) collection.
+
+    To get it we check all collections with a library override and check if
+    they have the loaded collection as their reference. If a collection is
+    not in the provided (known) override collections, we assume it's the newly
+    created one.
+    """
+    local_collections: set[bpy.types.Collection] = set()
+    for collection in get_overridden_collections_from_reference_collection(
+        loaded_collection
+    ):
+        if collection not in overridden_collections:
+            local_collections.add(collection)
+    if len(local_collections) != 1:
+        raise RuntimeError("Could not find the overridden collection.")
+
+    return local_collections.pop()
+
+
+def get_overridden_collections_from_reference_collection(
+    reference_collection: bpy.types.Collection,
+) -> Generator[bpy.types.Collection, None, None]:
+    """Get collections that are overridden versions of the reference collection.
+
+    Yields:
+        All collections that have an override library and have the
+        `reference_collection` collection as reference.
+    """
+    for collection in bpy.data.collections:
+        if not collection.override_library:
+            continue
+        if collection.override_library.reference == reference_collection:
+            yield collection
+
+
+def load_collection(
+    filepath,
+    link=True,
+    lib_container_name = None
+) -> bpy.types.Collection:
+    """Load a collection to the scene."""
+    with bpy.data.libraries.load(filepath, link=link, relative=False) as (
+        data_from,
+        data_to,
+    ):
+        if lib_container_name is None:
+            lib_container_name = [
+                instance for instance in data_from.collections
+                if instance == "AVALON_INSTANCES"
```

Comment:
This seems very wrong. I don't think it's _this_ collection we should enforce to load - it feels like the opposite of we want.
This `AVALON_INSTANCES` is the "loaded content" in a workfile publish (just by luck actually)...

I think we should work towards being more flexible and allowing to load any collection data so it's more universal and less restrictive. Maybe, we can provide some logic to "select" which container to load?